src/core/Dropdown/index.tsx

```diff
@@ -184,7 +194,9 @@ const Dropdown = <Multiple extends boolean | undefined = false>({
 
   function handleClickAway() {
     if (open) {
-      setOpen(false);
+      if (closeOnBlur && !shouldShowButtons) {
```

Comment:
I wonder if some comments would help in this file and functions to explain why we have certain conditions and diverged behavior, so we give readers enough context and information that will help them understand the code and make modifications down the road more confidently?

Reply:
Sure, I'll add some comments. 🤓